### PR TITLE
add --publid-addr --cert-file --key-file for teleport configure

### LINKED ISSUE #8989: Add `--public-addr`, `--cert-file` and `--key-file` flags to `teleport configure`
### What

Extend `teleport configure` command with the flags mentioned in the subject.

See the following PR comment for context: https://github.com/gravitational/teleport/pull/8988#discussion_r748895873.

The flags should set the following configuration fields in the `proxy_service` config:

| Flag | Config field |
| ---- | -------- |
| `--public-addr` | [Proxy.PublicAddr](https://github.com/gravitational/teleport/blob/v8.0.0-rc.2/lib/config/fileconf.go#L1057)  |
| `--cert-file` / `--key-file` | [Proxy.KeyPairs](https://github.com/gravitational/teleport/blob/v8.0.0-rc.2/lib/config/fileconf.go#L1070) |

Also:
- Update the snippet in documentation from the PR mentioned [above](https://github.com/gravitational/teleport/pull/8988#discussion_r748895873).
- Backport both code and documentation update to `branch/v8`.

### PR DESCRIPTION
**Pull request by greedy52**
closes #8989

Sample runs:
```
$ ./teleport configure --public-addr "a.com" --public-addr "b.com" --key-file "/tmp/key" --cert-file "/tmp/cert"
#
# A Sample Teleport configuration file.
# Creates a single proxy, auth and node server.
...
proxy_service:
  enabled: "yes"
  public_addr:
  - a.com
  - b.com
  https_keypairs:
  - key_file: /tmp/key
    cert_file: /tmp/cert
  acme: {}

$ ./teleport configure --public-addr "a.com" --key-file "/tmp/key" --cert-file "/tmp/cert" --acme --cluster-name my-cluster
ERROR: could not use --key-file/--cert-file when ACME is enabled

$ ./teleport configure --key-file "/tmp/key"
ERROR: please provide both --key-file and --cert-file

$ ./teleport configure --public-addr "a.com" --key-file "/tmp/key" --cert-file "/tmp/cert"
ERROR: failed to load x509 key pair from --key-file and --cert-file
tls: failed to find any PEM data in certificate input

### CODE REVIEW

File: lib/config/fileconf.go
Comment (r0mant) [inline lib/config/fileconf.go:160]:
```suggestion
			return nil, trace.BadParameter("cannot use --key-file/--cert-file when ACME is enabled")
```

File: lib/config/fileconf.go
Comment (r0mant) [inline lib/config/fileconf.go:157]:
```suggestion
			return nil, trace.BadParameter("please provide --cluster-name when using ACME, for example --cluster-name=example.com")
```

File: tool/teleport/common/teleport.go
Comment (r0mant) [inline tool/teleport/common/teleport.go:231]:
```suggestion
	dump.Flag("cert-file", "Path to a TLS certificate file for the proxy.").ExistingFileVar(&dumpFlags.CertFile)
	dump.Flag("key-file", "Path to a TLS key file for the proxy.").ExistingFileVar(&dumpFlags.KeyFile)
```

File: tool/teleport/common/teleport.go
Comment (r0mant) [inline tool/teleport/common/teleport.go:372]: Can we figure out the path ourselves using something like `filepath.Abs` instead of asking the user? Same for cert file.

File: lib/config/fileconf.go
Comment (r0mant) [inline lib/config/fileconf.go:217]: Do we validate somewhere that these files actually exist, can be opened and that they form a valid keypair?

Comment (greedy52) [reply lib/config/fileconf.go:217]: It's validated in `teleport` binary through `ExistingFileVar` flags
```
    dump.Flag("cert-file", "Path to a TLS certificate file").ExistingFileVar(&dumpFlags.CertFile)                                
    dump.Flag("key-file", "Path to a TLS key file").ExistingFileVar(&dumpFlags.KeyFile) 
```

Comment (r0mant) [reply lib/config/fileconf.go:217]: @greedy52 Cool. Let's also make sure that they are a valid key pair (e.g. using `tls.LoadX509KeyPair`).

File: tool/teleport/common/teleport.go
Comment (greedy52) [reply tool/teleport/common/teleport.go:372]: will give it a try

**Review by r0mant — APPROVED** [review]
@greedy52 lgtm, congrats on your first PR!

Can you please backport this to branch/v8 as well?

**Comment by greedy52** [comment]
> @greedy52 lgtm, congrats on your first PR!
> 
> Can you please backport this to branch/v8 as well?

#9049

**Comment by r0mant** [comment]
@quinqu This is weird, the Github bot requested a review from me after I approved this PR for some reason (see above). Do you know why it happened?

**Review by r0mant — APPROVED** [review]

**Comment by r0mant** [comment]
@quinqu And same thing just happened again ^

Looks like the bot is dismissing reviews for some reason.

File: lib/config/fileconf.go
Comment (smallinsky) [inline lib/config/fileconf.go:162]: nit: It would be nice to introduce a separate function doing only `SampleFlags` flags validation.

**Review by smallinsky — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**
